### ISSUE TITLE
[BUG] check-unused-files reporting entire project unused

### ISSUE THREAD

**Issue by tomgilder** [dart-code-checker/dart-code-metrics#850]
* **Dart code metrics version:** 4.15.1
* **Dart sdk version:** 2.17.0

**Please show your full configuration:**

None - just using `check-unused-files `

**What did you do? Please include the source code example causing the issue.**

Ran `flutter pub run dart_code_metrics:metrics check-unused-files lib`

**What did you expect to happen?**

Unused files found

**What actually happened?**

All 227 Dart files are reported as unused.

Our entry point files are listed as unused, I assume that's the cause of the problem:

```
⚠ unused file: lib/main.dart
⚠ unused file: lib/main_dev.dart
⚠ unused file: lib/main_mock.dart
⚠ unused file: lib/main_test.dart
```

I'm afraid this is a commercial project that we can't share, but are there any logs I can provide to help debugging?

**Comment by tomgilder** [comment]
Submitted the bug and immediately found the issue 😁 

We have a `analysis_options.yaml` file in a subfolder because we need to disable strong-mode for some generated code:

```
analyzer:
  strong-mode:
    implicit-dynamic: true
```

Deleting that file fixes the problem.

I guess we can work around this by excluding all of that code from analysis, but should this be supported?

**Comment by incendial** [comment]
That's interesting, can you please provide an example repo with this issue reproduced? I think I saw the same approach with `analysis_options.yaml` in the another issue, but nothing about this problem

**Comment by incendial** [comment]
I was able to reproduce the problem. It's definitely an issue and looks like we have double analysis for such cases.

**Comment by incendial** [comment]
Available in 4.16.0! 🚀
